byterun/floats.c

```diff
@@ -530,6 +650,49 @@ CAMLprim value caml_log1p_float(value f)
   return caml_copy_double(caml_log1p(Double_val(f)));
 }
 
+#ifndef HAS_C99_FLOAT_OPS
+static inline double simple_erf(double x)
+{
+  double a1 =  0.254829592, a2 = -0.284496736, a3 = 1.421413741,
+         a4 = -1.453152027, a5 =  1.061405429, p  = 0.3275911,
+         t, y;
+
+  int sign = (x >= 0) ? 1 : -1;
+  x = fabs(x);
+  t = 1.0 / (1.0 + p * x);
+  y = 1.0 - (((((a5 *t  + a4) * t) + a3) * t + a2) * t + a1) * t * exp(-x * x);
```

Comment:
Sometimes `0` is just simply the closest answer even a good approximation can come up with.  The C99 implementation on my platform returns `1` or `-1`, too, whenever  `|x| >= ~6`.  The implemented approximation above was published in one of the most widely used reference books (Abramowitz and Stegun: Handbook of Mathematical Functions).  More modern approximations are more accurate, but typically a hell lot more complicated.  Note, too, that some global approximations, which have a guaranteed maximum error in the whole range, may perform much more poorly in ranges that actually matter to people.  I really don't think we need to go overboard here.  To me this is good enough for a fallback solution.